### PR TITLE
update web3sdk dev

### PR DESCRIPTION
**Pull request by ywy2090**
1. Initialize the connection node timeout to milliseconds
2. fix load function in P12Manager.java for NullPointException
3. remove Encode , Decode files
4. format few java files.
5. add version number for the channel protocol
6. fixed a problem with dynamic array input arrays not being empty

### CODE REVIEW

File: src/main/java/org/fisco/bcos/channel/client/Service.java [hunk 1]
```diff
@@ -915,6 +936,167 @@ public void run() {
...
+            }
+
+            ByteBuf out = ctx.alloc().buffer();
+            response.writeHeader(out);
+            response.writeExtra(out);
+
+            ctx.writeAndFlush(out);
+        } else if ("1".equals(content)) {
+            logger.trace("heartbeat response");
+        }
```
Comment (wheatli) [inline src/main/java/org/fisco/bcos/channel/client/Service.java]: plz give handling for default branch

File: src/main/java/org/fisco/bcos/channel/client/Service.java [hunk 2]
```diff
@@ -915,6 +936,167 @@ public void run() {
...
+    }
+
+    public org.fisco.bcos.channel.protocol.Version getUsageChlVersion(ChannelHandlerContext ctx) {
```
Comment (wheatli) [inline src/main/java/org/fisco/bcos/channel/client/Service.java]: the function name is ugly, plz do not use an abbreviation, getChannelVersion is more available

File: src/main/java/org/fisco/bcos/channel/protocol/parser/BlkNotifyParser.java [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+
+import com.fasterxml.jackson.core.JsonParseException;
+import com.fasterxml.jackson.databind.JsonMappingException;
+import java.io.IOException;
+import java.math.BigInteger;
+import org.fisco.bcos.channel.dto.BcosBlkNotify;
+import org.fisco.bcos.channel.protocol.Version;
+import org.fisco.bcos.web3j.protocol.ObjectMapperFactory;
+
+public class BlkNotifyParser {
```
Comment (wheatli) [inline src/main/java/org/fisco/bcos/channel/protocol/parser/BlkNotifyParser.java]: Do not use an abbreviation asap, BlockNotificationParser is more available

File: src/main/java/org/fisco/bcos/web3j/codegen/SolidityFunctionWrapper.java [hunk 4]
```diff
@@ -131,9 +131,15 @@ void generateJavaFiles(
             String destinationDir,
             String basePackageName,
             Map<String, String> addresses)
-            throws IOException, ClassNotFoundException {
+            throws IOException, ClassNotFoundException, UnsupportedOperationException {
         String className = Strings.capitaliseFirstLetter(contractName);
 
+        if (bin.length() > 0x40000) {
```
Comment (wheatli) [inline src/main/java/org/fisco/bcos/web3j/codegen/SolidityFunctionWrapper.java]: Do not use a magic number in the code, define it as constant

File: src/main/java/org/fisco/bcos/channel/client/Service.java [hunk 5]
```diff
@@ -915,6 +936,167 @@ public void run() {
...
+            // logger.trace(" heartbeat packet, heartbeat is {} ", bcosHeartbeat);
+            int heartBeat = bcosHeartbeat.getHeartBeat();
+            content = String.valueOf(heartBeat);
+        } catch (UnsupportedEncodingException e) {
+            logger.error("heartbeat packet cannot be parsed, data is {}", data);
+        } catch (Exception e) {
+            logger.error("heartbeat packet exception, data is {}", data);
+        }
+
+        if ("0".equals(content)) {
```
Comment (wheatli) [inline src/main/java/org/fisco/bcos/channel/client/Service.java]: Can we define the 0/1 as a type, make the code readable and maintainable

File: src/main/java/org/fisco/bcos/channel/client/Service.java [hunk 6]
```diff
@@ -915,6 +936,167 @@ public void run() {
...
+                    org.fisco.bcos.channel.protocol.Version.getHighestSupported();
+
+            logger.trace(
+                    " remote host is {}, node channel version info is {}, node highest version is {}, sdk highest version is {}",
+                    remoteEndPoint,
+                    nodeVersion,
+                    nodeHightestChlVersion.getVersionNumber(),
+                    SDKHighestVersion.getVersionNumber());
+
+            if (nodeHightestChlVersion.getVersionNumber() > SDKHighestVersion.getVersionNumber()) {
```
Comment (wheatli) [inline src/main/java/org/fisco/bcos/channel/client/Service.java]: plz give some comments for the selection logic

File: src/main/java/org/fisco/bcos/channel/dto/BcosBlkNotify.java [hunk 7]
```diff
@@ -0,0 +1,36 @@
+package org.fisco.bcos.channel.dto;
+
+import com.fasterxml.jackson.annotation.JsonIgnore;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import java.math.BigInteger;
+
+public class BcosBlkNotify {
```
Comment (wheatli) [inline src/main/java/org/fisco/bcos/channel/dto/BcosBlkNotify.java]: the class name should be a noun, and all codes belong to bcos, so you don't have to use bcos as a prefix
maybe BlockNotification is more available

Comment (ywy2090) [reply src/main/java/org/fisco/bcos/channel/dto/BcosBlkNotify.java]: ok, It has been modified to BcosBlockNotification，Bcos prefix is just use the same style as others .

File: src/main/java/org/fisco/bcos/channel/client/Service.java (hunk 6 above)
Comment (ywy2090) [reply src/main/java/org/fisco/bcos/channel/client/Service.java]: > plz give some comments for the selection logic

This logic is droped

File: src/main/java/org/fisco/bcos/channel/client/Service.java (hunk 5 above)
Comment (ywy2090) [reply src/main/java/org/fisco/bcos/channel/client/Service.java]: > Can we define the 0/1 as a type, make the code readable and maintainable

Old logic keeps the same, new protocol already define as 0/1

File: src/main/java/org/fisco/bcos/web3j/codegen/SolidityFunctionWrapper.java (hunk 4 above)
Comment (ywy2090) [reply src/main/java/org/fisco/bcos/web3j/codegen/SolidityFunctionWrapper.java]: > Do not use a magic number in the code, define it as constant

done

File: src/main/java/org/fisco/bcos/channel/client/Service.java (hunk 2 above)
Comment (ywy2090) [reply src/main/java/org/fisco/bcos/channel/client/Service.java]: > the function name is ugly, plz do not use an abbreviation, getChannelVersion is more available
done

File: src/main/java/org/fisco/bcos/channel/protocol/parser/BlkNotifyParser.java (hunk 3 above)
Comment (ywy2090) [reply src/main/java/org/fisco/bcos/channel/protocol/parser/BlkNotifyParser.java]: > BlockNotificationParser

done

File: src/main/java/org/fisco/bcos/channel/client/Service.java [hunk 8]
```diff
@@ -915,6 +939,162 @@ public void run() {
...
+        logger.info(
+                "connection established，send sdk channel supported highest protocol to the connection , seq: {}, channel handshake: {}",
+                message.getSeq(),
+                channelHandshake);
+
+        ByteBuf out = ctx.alloc().buffer();
+        message.writeHeader(out);
+        message.writeExtra(out);
+
+        ctx.writeAndFlush(out);
```
Comment (morebtcg) [inline src/main/java/org/fisco/bcos/channel/client/Service.java]: Wait the node's response

File: src/main/java/org/fisco/bcos/channel/protocol/EnumNodeVersion.java [hunk 9]
```diff
@@ -0,0 +1,30 @@
+package org.fisco.bcos.channel.protocol;
+
+public enum EnumNodeVersion {
+    BCOS_RC1("2.0.0-rc1"),
+    BCOS_RC2("2.0.0-rc2"),
+    BCOS_RC3("2.0.0-rc3"),
+    BCOS_20("2.0.0"),
```
Comment (bxq2011hust) [inline src/main/java/org/fisco/bcos/channel/protocol/EnumNodeVersion.java]:
```suggestion
    BCOS_2_0_0("2.0.0"),
```

**Pull request merged**